### ISSUE TITLE
Can it run on SGX544MP3 on ODROID XU？

### ISSUE THREAD

**Issue by yangchaohong** [openpvrsgx-devgroup/linux_openpvrsgx#10]
Can it run on SGX544MP3 on ODROID XU(Exynos 5410)？
Some Allwinner chips use SGX544,can they use the same DDK?
How to deal with blobs?

**Comment by goldelico** [comment]
> Am 06.04.2024 um 15:55 schrieb yangchaohong ***@***.***>:
> 
> 
> Can it run on SGX544MP3 on ODROID XU(Exynos 5410)？
> Some Allwinner chips use SGX544,can they use the same DDK?
> How to deal with blobs?

In theory yes, but in practise...

This educated opinion comes from the observation that this SoC seems to be ARM Cortex A15
like the OMAP5. One difference is that the Exynos uses 3 GPU cores while the OMAP5 only has 2.

As far as I remember there is some #define in the kernel driver code to set the number of
GPU cores handled by the driver. But what I can't tell is if the user-space takes care of it.

So there is a chance to get the driver adapted.

And it might be possible to load the OMAP5 DDK 1.17 pvrsrvctl (which contains and downloads
the GPU firmware). It might also need a hack to not check for the CPU type.

So as a starting point you might look at:

https://github.com/openpvrsgx-devgroup/linux_openpvrsgx/commit/17012dc1527a9139b7fecae8fb5f7a5267e33939

and

https://github.com/openpvrsgx-devgroup/linux_openpvrsgx/commit/b0506f5756f9f6cd64f90d30abf87d9007c7074d

None of this code has been tested since we are more an archive than an active project...

But with user-space I think the TI provided binary libraries are too much tied to their chips.

Best,
Nikolaus

**Comment by yangchaohong** [comment]
But HDMI in Exynos5410 seems doesn't work in the mainline kernel.It seems impossible for me to make it.

**Comment by goldelico** [comment]
> Am 07.04.2024 um 16:05 schrieb yangchaohong ***@***.***>:
> 
> 
> But HDMI in Exynos5410 seems doesn't work in the mainline kernel.It seems impossible for me to make it.


Yes, without having some video output interface first it doesn't make much sense.

**Comment by mirh** [comment]
https://forum.odroid.com/viewtopic.php?t=41139
https://forum.odroid.com/viewtopic.php?t=40927
@gabrielribbae
